### ISSUE TITLE
Formly not able to match the correct option in oneOf field of JsonSchema

### ISSUE THREAD

**Issue by dev-kanishk** [ngx-formly/ngx-formly#3723]
**Description**
oneOF field is unable to match the correct option in case of nested oneOf and string input when rendered with the data (model)

Model:
```
{ heading: 'custom heading' }
```

Json schema:

```
{
        type: 'object',
        properties: {
          heading: {
            title: 'Heading',
            oneOf: [
              {
                title: 'Option 1: Standard heading',
                type: 'string',
                oneOf: [
                  { const: 'Heading 1' },
                  { const: 'Heading 2' },
                  { const: 'Heading 3' },
                ],
              },
              {
                title: 'Option 2: Custom heading',
                type: 'string',
              },
            ],
          },
        },
      }
```

**Minimal Reproduction**

Use the following the stackblitz:

https://stackblitz.com/edit/angular-kuxmzz?file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.module.ts

Issue:
Form should have been rendered with "Option 2: Custom heading" as selected option of oneOf and input field should be populated with "custom heading"

Formly version: 6.1.8

**Comment by rishabh413** [comment]
Hi all

Any pointers for this?

**Comment by dev-kanishk** [comment]
Hey, Thank for fixing the issue. the example I provided is working now after the fix. But facing a similar issue, The following schema form is rendered but on filling the data model is not updated.

```
{
  "type": "object",
  "properties": {
    "options": {
      "title": "Options",
      "type": "array",
      "items" : {
      "oneOf" : [
          {
            "title" : "Option 1", 
            "enum" : [ "10", "Heading 2" ]
          },
          {
            "title" : "Option 2", 
            "type" : "number" 
          }
        ]
      }
    }
  }
}
```

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.2.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.2.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
